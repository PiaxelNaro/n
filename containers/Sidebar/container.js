// @flow
import React from 'react';
import classnames from 'classnames';
import styled from 'styled-components';
import Logo from '../../components/Logo';
import FlexGrid from '../../components/FlexGrid';

const Wrapper = styled.div`
    width: ${props => (props.isCollapsable ? '45px' : '200px')};
    float: left;
`;

const LogoLink = styled.a`
    display: flex;
    padding-left: 15px;
    &.is-collapsable {
        padding-left: 6px;
    }
    &.is-collapsable.is-hover {
        padding-left: 15px;
    }
    align-items: center;
    overflow: hidden;
    text-decoration: none;
    color: inherit;
`;

const LogoDefault = styled(FlexGrid)`
    max-width: 400px;
    align-items: flex-end;
    color: #00bfaa;
    font-size: 9px;
    opacity: 1;
    visibility: visible;
    overflow: hidden;
    &.is-collapsable {
        max-width: 0;
        opacity: 0;
        visibility: hidden;
        transition: opacity .2s, visibility .2s; /* hide */
    }
    &.is-collapsable.is-hover {
        max-width: 400px;
        opacity: 1;
        visibility: visible;
        transition: opacity .25s .2s, visibility .25s .2s; /* show */
    }
`;

const LogoMini = styled(FlexGrid)`
    max-width: 0;
    opacity: 0;
    visibility: hidden;
    overflow: hidden;
     &.is-collapsable {
         max-width: 100%;
         opacity: 1;
         visibility: visible;
         transition: opacity .25s .2s, visibility .25s .2s;
    }
    &.is-collapsable.is-hover {
        max-width: 0;
        opacity: 0;
        visibility: hidden;
        transition: opacity .2s, visibility .2s;     /* hide */
    }
`;

const Main = styled.div`
    position: fixed;
    top: 0;
    left: 0;
    width: 200px;
    height: 100%;
    float: left;
    background-color: #000;
    color: #FFF;
    font-size: 13px;
    overflow-y: auto;
    overflow-x: hidden;
    &.is-collapsable {
        width: 45px;
        transition: width .25s .2s;
        z-index: 1002;
    }
    &.is-collapsable.is-hover {
        width: 200px;
        transition: width .25s .2s;
        z-index: 1001;
    }
`;
const Content = styled.div`
    opacity: 1;
    visibility: visible;
    &.is-collapsable {
        opacity: 0;
        visibility: hidden;
        transition: opacity .25s .2s, visibility .25s .2s;
    }
    &.is-collapsable.is-hover {
        opacity: 1;
        visibility: visible;
    }
`;

const Mask = styled.div`
    height: 100%;
    width: 100%;
    position: fixed;
    top: 0;
    left: -9999px;
    z-index: 1000;
    opacity: 0;
    background-color: #000;
    transition: opacity .2s, background-color .2s;
    transition-delay: .25s;
    &.is-collapsable.is-hover {
        left: 0;
        opacity: .3;
    }
`;

const PremiumText = styled.div`
    margin: 0 0 5px 2px;
    color: #00bfaa;
`;

class Container extends React.Component {
    constructor() {
        super();
        this.state = {
            isHover: false,
        };
        this.onMouseEnter = this.onMouseEnter.bind(this);
        this.onMouseLeave = this.onMouseLeave.bind(this);
    }

    props: {
        link: string,
        isCollapsable: boolean,
        isPremium: boolean,
        children?: any,
    };

    state: {
        isHover: boolean,
    };

    onMouseEnter: Function;
    onMouseLeave: Function;

    onMouseEnter() {
        this.setState({
            isHover: true,
        });
    }

    onMouseLeave() {
        this.setState({
            isHover: false,
        });
    }

    render() {
        const { link, isCollapsable, isPremium, children } = this.props;
        const classes = classnames({
            'is-collapsable': isCollapsable,
            'is-hover': this.state.isHover,
        });

        let mask;
        if (isCollapsable) {
            mask = <Mask className={classes} />;
        }

        let premiumText;
        if (isPremium) {
            premiumText = <PremiumText>PREMIUM</PremiumText>;
        }
        return (
            <Wrapper isCollapsable={isCollapsable}>
                <Main
                    className={`AASidebar ${classes}`}
                    onMouseEnter={this.onMouseEnter}
                    onMouseLeave={this.onMouseLeave}
                >

                    <LogoLink href={link} className={classes}>
                        <LogoMini className={classes}>
                            <Logo mini premium={isPremium} />
                        </LogoMini>
                        <LogoDefault className={classes}>
                            <Logo />
                            {premiumText}
                        </LogoDefault>
                    </LogoLink>

                    <Content className={classes}>
                        {children}
                    </Content>
                </Main>
                {mask}
            </Wrapper>
        );
    }
}

export default Container;
