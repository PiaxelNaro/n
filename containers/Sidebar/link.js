// @flow
import React from 'react';
import classnames from 'classnames';
import styled from 'styled-components';
import LockIcon from 'react-icons/lib/md/lock-outline';
import AAIcon from '../../components/AAIcon';
import FlexGrid from '../../components/FlexGrid';

const Wrapper = styled.a`
    display: block;
    padding: 5px 15px;
    text-decoration: none;
    cursor: pointer;
    color: #F2F2F2;
    &:hover,
    &:focus {
        background-color: #233140;
    };
    &.is-active {
        background-color: #2f669f;
    };
`;

const Version = styled.div`
    margin: 0 0 0 5px;
    font-size: 9px;
    color: #3f88d4;
    text-transform: uppercase;
`;

const Link = ({
    onClick,
    icon,
    isActive = false,
    isBeta,
    isNew,
    isLocked,
    isConnected,
    children,
}: {
    onClick?: () => void,
    icon?: any,
    isActive?: boolean,
    isBeta?: boolean,
    isNew?: boolean,
    isLocked?: boolean,
    isConnected?: boolean,
    children?: any,
}) => {
    let _onClick = null;
    if (!isLocked) {
        _onClick = onClick;
    }
    const classes = classnames({
        'is-active': isActive,
    });

    return (
        <Wrapper className={classes} onClick={_onClick}>
            <FlexGrid gutter="2px">
                {icon}
                <FlexGrid.Item>
                    <FlexGrid>
                        {children}
                        <Version>
                            {isBeta ? 'beta' : ''}
                            {isNew ? 'new' : ''}
                        </Version>
                    </FlexGrid>
                </FlexGrid.Item>
                {isLocked ? <LockIcon size="14" /> : ''}
                {isConnected ? <AAIcon name="aa-connections" /> : ''}
            </FlexGrid>
        </Wrapper>
    );
};

export default Link;
