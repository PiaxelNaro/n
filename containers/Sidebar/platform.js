// @flow
import React from 'react';
import classnames from 'classnames';
import styled from 'styled-components';
import AAIcon from '../../components/AAIcon';

import Text from '../../components/Text';
import Link from './link';

export type NavItem = {
    id: string,
    title: string,
    href?: string,
    isNew?: boolean,
    isBeta?: boolean,
    isLocked?: boolean,
    isConnected?: boolean,
    isVisible?: boolean,
};

const GroupItem = styled(Text)`
    padding-left: 15px;
`;

const NavGroupWrapper = styled.div`
    margin: 0 0 10px;
`;

const NavGroupTitle = styled(Text)`
    margin: 10px 0 0;
    padding: 5px 15px 5px 30px;
    color: #818A94;
    text-transform: uppercase;
    font-size: 10px;
`;

export const NavGroup = ({
    title,
    current,
    items,
    onClick,
}: {
    title: string,
    current: string,
    items: Array<NavItem>,
    onClick: any => void,
}) => {
    let GroupTitle;
    if (title) {
        GroupTitle = (
            <NavGroupTitle ellipsis className="AASidebar-group-title">
                {title}
            </NavGroupTitle>
        );
    }

    return (
        <NavGroupWrapper className="AASidebar-group">
            {GroupTitle}
            {items.map(item =>
                <Link
                    key={item.title}
                    isActive={current === item.id}
                    isNew={item.isNew}
                    isBeta={item.isBeta}
                    isConnected={item.isConnected}
                    isLocked={item.isLocked}
                    onClick={() => onClick(item.href)}
                >
                    <GroupItem ellipsis>
                        {item.title}
                    </GroupItem>
                </Link>
            )}
        </NavGroupWrapper>
    );
};

const Wrapper = styled.div`
    margin: 0 0 10px 0;
`;

const Content = styled.div`
    opacity: 1;
    max-height: 1000px;
    overflow: hidden;
    visibility: visible;
    transition: all 0.2s ease-in;
    &.collapse {
        max-height: 0;
        opacity: 0;
        visibility: hidden;
        transition: all 0.2s ease-in;
    }
`;

const PlatformTitle = styled(Text)`
    color: #818a94;
    text-transform: uppercase;
    font-size: 12px;
`;

class Platform extends React.Component {
    constructor() {
        super();
        this.state = {
            collapse: false,
        };
        this.toggleCollapse = this.toggleCollapse.bind(this);
    }

    props: {
        current: string,
        title: string,
        navGroup: Array<{ id: string, title: string, items: Array<NavItem> }>,
        onClick: string => void,
    };

    state: {
        collapse: boolean,
    };

    toggleCollapse: Function;

    toggleCollapse() {
        this.setState({
            collapse: Boolean(!this.state.collapse),
        });
    }

    render() {
        const { title, navGroup } = this.props;
        const classes = classnames({
            collapse: this.state.collapse,
        });

        let arrowIcon = <AAIcon name="aa-caret-down" />;
        if (this.state.collapse) {
            arrowIcon = <AAIcon name="aa-caret-right" />;
        }

        return (
            <Wrapper className={`AASidebar-platform ${classes}`}>
                <Link
                    icon={arrowIcon}
                    className="AASidebar-platform-title"
                    onClick={this.toggleCollapse}
                >
                    <PlatformTitle ellipsis>{title}</PlatformTitle>
                </Link>
                <Content className={`AASidebar-platform-content ${classes}`}>
                    {navGroup &&
                        navGroup.map(group =>
                            <NavGroup
                                key={group.id}
                                title={group.title}
                                current={this.props.current}
                                items={group.items}
                                onClick={this.props.onClick}
                            />
                        )}
                </Content>
            </Wrapper>
        );
    }
}

export default Platform;
NavGroup.displayName = 'NavGroup';
