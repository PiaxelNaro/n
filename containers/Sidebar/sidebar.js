// @flow
import React from 'react';
import AAIcon from '../../components/AAIcon';
import Text from '../../components/Text';
import Container from './container';
import Link from './link';
import Platform from './platform';

const Sidebar = ({
    home,
    platforms,
    current,
    isCollapsable = false,
    isPremium = false,
    onClickNav,
}: {
    home: any,
    platforms: any,
    current: string,
    isCollapsable?: boolean,
    isPremium?: boolean,
    onClickNav: string => void,
}) =>
    <Container link={home.href} isCollapsable={isCollapsable} isPremium={isPremium}>
        <Link
            link={home.href}
            icon={<AAIcon name="aa-home" />}
            isActive={current === home.id}
        >
            <Text ellipsis>
                {home.title}
            </Text>
        </Link>

        {platforms.map(platform => {
            if (!platform.items) {
                return (
                    <Link
                        key={platform.id}
                        link={platform.href}
                        icon={<AAIcon name={platform.icon} />}
                        isActive={current === platform.id}
                    >
                        <Text ellipsis textTransform="uppercase">
                            {platform.title}
                        </Text>
                    </Link>
                );
            }
            return (
                <Platform
                    current={current}
                    key={platform.id}
                    title={platform.title}
                    navGroup={platform.items}
                    onClick={onClickNav}
                />
            );
        })}
    </Container>;

export default Sidebar;
