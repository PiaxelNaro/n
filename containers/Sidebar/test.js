// @flow
import NextRouter from 'next/router';
import { mapStateToProps } from '.';
import mockI18n from '../../modules/i18n/mockI18n';

const Router: any = NextRouter;

jest.mock('next/router', () => ({
    pathname: 'test',
    push: jest.fn(),
}));

describe('<Sidebar />', () => {
    it('connects Sidebar to the redux store', () => {
        const props = mapStateToProps(
            {
                user: {
                    user: {
                        isPremium: true,
                        isANA: false,
                        isINT: false,
                    },
                },
            },
            {
                isCollapsable: true,
                i18n: mockI18n,
            }
        );

        expect(props.home.href).toEqual('/dashboard/home/');

        const href = '/home';
        props.onClickNav(href);
        expect(Router.push).toHaveBeenCalledWith(href);
    });
});
