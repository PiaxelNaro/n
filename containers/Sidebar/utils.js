// @flow
import { createSelector } from 'reselect';
import { type I18nType } from '../../modules/i18n';
import { type UserType } from '../../modules/user';

// If user is Non-ANA, empty sub navs
const getANASidebar = (user: UserType, i18n: I18nType) => {
    if (user.isANA) {
        return [
            {
                id: 'c',
                title: '',
                items: [
                    {
                        id: 'c1',
                        title: i18n.gettext('帮助'),
                        href: '/help/',
                        isVisible: true,
                    },
                    {
                        id: 'c2',
                        title: i18n.gettext('课程'),
                        href: '/course',
                        isVisible: true,
                    },
                ],
            },
        ];
    }
    return null;
};

// Store PP locked when
// 1. user does not buy any store contract
// 2. user only buy store PYA or store PYP
const isStorePPLocked = (user: UserType) =>
    !user.isINT || !user.intAuth.store || !user.intAuth.store.pp;

// Store PYA invisible when
// 1. user does not buy any store contract
// 2. user is store PP only or store PYP only
const isStorePYAUser = (user: UserType) =>
    user.isINT && user.intAuth.store && user.intAuth.store.pya;

// Store PYP visible when
// user is store PYA or store PYP
const isStorePYPVisible = (user: UserType) =>
    user.isINT &&
    user.intAuth.store &&
    (user.intAuth.store.pyp || user.intAuth.store.pya);

// Store PYP locked when
// 1. user does not buy store contract
// 2. user is not store PYP
const isStorePYPLocked = (user: UserType) =>
    !user.isINT || !user.intAuth.store || !user.intAuth.store.pyp;

// Usage PP unlocked when user buy usage pp
const isUsagePPLocked = (user: UserType) =>
    !user.isINT || !user.intAuth.usage || !user.intAuth.usage.pp;

// Usage PYA show and unlocked if user buy usage pya
const isUsagePYAUser = (user: UserType) =>
    user.isINT && user.intAuth.usage && user.intAuth.usage.pya;

// Usage retention unlocked if user buy usage pp
const isUsagePPUser = (user: UserType) =>
    user.isINT && user.intAuth.usage && user.intAuth.usage.pp;

const isMarketAsoUser = (user: UserType) =>
    user.isINT && user.intAuth.marketing && user.intAuth.marketing.aso;

const isMarketAdvertisingUser = (user: UserType) =>
    user.isINT && user.intAuth.marketing && user.intAuth.marketing.advertising;

const getINTSidebar = (user: UserType, i18n: I18nType) =>
    [
        {
            id: 'a',
            title: i18n.gettext('排课'),
            items: [
                {
                    id: 'a1',
                    title: i18n.gettext('选课'),
                    href: '/#',
                    isVisible: true,
                    isLocked: !user.isINT,
                },
            ],
        },
        {
            id: 'b',
            title: i18n.gettext('查询'),
            items: [
                {
                    id: 'b1',
                    title: i18n.gettext('课程表'),
                    href: '/apps/ios/top-chart/',
                    isVisible: true,
                },
            ],
        }
    ].map(child => ({
        ...child,
        items: child.items.filter(item => item.isVisible),
    }));

const getSidebar = createSelector(
    user => user,
    (user, i18n) => i18n,
    (user: UserType, i18n: I18nType) => {
        const ANA_NAV = getANASidebar(user, i18n);
        const INT_NAV = getINTSidebar(user, i18n);
        return {
            current: 'a1',
            home: {
                id: 'home',
                title: i18n.gettext('首页'),
                href: '/home',
                isVisible: true,
            },
            platforms: [
                {
                    id: 'my_app_analytics',
                    title: i18n.gettext('学生功能'),
                    href: '/f',
                    icon: 'aa-connections',
                    items: ANA_NAV,
                },
                {
                    id: 'market_intelligence',
                    title: i18n.gettext('管理员功能'),
                    items: INT_NAV,
                },
            ],
        };
    }
);

export default getSidebar;
