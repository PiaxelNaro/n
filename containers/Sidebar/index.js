// @flow
import { connect } from 'react-redux';
import Router from 'next/router';

import { translate, type I18nType } from '../../modules/i18n';
import getUser from '../../store/User/selectors';
import Sidebar from './sidebar';
import getSidebar from './utils';

const onClickNav = (url: string) => Router.push(url);

const mapStateToProps = (
    state: Object,
    props: { i18n: I18nType, isCollapsable: boolean }
) => {
    const user = getUser(state);
    const navs = getSidebar(user, props.i18n);
    return {
        home: navs.home,
        platforms: navs.platforms,
        isPremium: user.isAdmin,
        onClickNav,
    };
};

export default translate(connect(mapStateToProps)(Sidebar));
export { mapStateToProps };
