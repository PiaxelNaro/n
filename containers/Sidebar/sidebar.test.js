import React from 'react';
import { shallow } from 'enzyme';
import Container from './container';
import Link from './link';
import Platform, { NavGroup } from './platform';
import Sidebar from './sidebar';

describe('<Link>', () => {
    it('render sidebar link', () => {
        const spy = jest.fn();
        const link = shallow(<Link link="/dashboard" icon={<div />} onClick={spy} />);
        expect(link).toMatchSnapshot();
        expect(link.hasClass('is-active')).toBeFalsy();

        link.prop('onClick')();
        expect(spy).toHaveBeenCalled();

        link.setProps({
            isBeta: true,
            isNew: true,
            isActive: true,
            isLocked: true,
            isConnected: true,
        });
        link.update();
        expect(link).toMatchSnapshot();
        expect(link.hasClass('is-active')).toBeTruthy();
    });
});

describe('<Container>', () => {
    it('render container', () => {
        const selector = shallow(
            <Container link="/home">
                test
            </Container>
        );
        expect(selector).toMatchSnapshot();

        selector.setProps({
            isCollapsable: true,
            isPremium: true,
        });
        selector.update();
        expect(selector).toMatchSnapshot();

        selector.find('.AASidebar').prop('onMouseEnter')();
        expect(selector.find('.AASidebar').hasClass('is-hover')).toBe(true);
        selector.find('.AASidebar').prop('onMouseLeave')();
        expect(selector.find('.AASidebar').hasClass('is-hover')).toBe(false);
    });

    it('render container wrapper', () => {
        const selector = shallow(
            <Container link="home">
                test
            </Container>
        );
        expect(selector.dive()).toMatchSnapshot();

        selector.dive().setProps({
            isCollapsable: true,
        });
        expect(selector.dive()).toMatchSnapshot();
    });
});

describe('<NavGroup>', () => {
    it('render NavGroup with title', () => {
        const spy = jest.fn();
        const title = 'App Store Rankings';
        const itemList = [
            {
                id: 'my_app_analytics_overview',
                title: 'Overview',
                href: '/comparator/',
            },
        ];
        const selector = shallow(
            <NavGroup title={title} items={itemList} onClick={spy} />
        );
        expect(selector).toMatchSnapshot();
        expect(selector.find('.AASidebar-group-title').length).toEqual(1);

        selector.find('Link').at(0).prop('onClick')();
        expect(spy).toHaveBeenCalledWith('/comparator/');
    });

    it('render NavGroup without header', () => {
        const itemList = [
            {
                id: 'my_app_analytics_overview',
                title: 'Overview',
                href: '/comparator/',
            },
        ];
        const selector = shallow(<NavGroup title="" items={itemList} />);
        expect(selector).toMatchSnapshot();
        expect(selector.find('.AASidebar-group-title').length).toEqual(0);
    });
});

describe('<Platform>', () => {
    it('render', () => {
        const navGroup = [
            {
                id: 'analytics_navs',
                title: '',
                items: [
                    {
                        id: 'my_app_analytics_overview',
                        title: 'Overview',
                        href: '/comparator/',
                    },
                ],
            },
        ];
        const spy = jest.fn();
        const selector = shallow(
            <Platform
                title=""
                current="my_app_analytics_overview"
                navGroup={navGroup}
                onClick={spy}
            />
        );
        expect(selector).toMatchSnapshot();

        // test collapse and expand sidebar
        expect(selector.hasClass('collapse')).toBeFalsy();
        selector.find('.AASidebar-platform-title').simulate('click');
        selector.update();
        expect(selector.hasClass('collapse')).toBeTruthy();
        selector.find('.AASidebar-platform-title').simulate('click');
        expect(selector.hasClass('collapse')).toBeFalsy();

        // test onClick
        selector.find('NavGroup').at(0).prop('onClick')();
        expect(spy).toHaveBeenCalled();
    });
});

describe('<Sidebar>', () => {
    it('render sidebar', () => {
        const spy = jest.fn();
        const platforms = [
            {
                id: 'my_app_analytics',
                title: 'My app analytics',
                href: '/comparator/landing/',
                items: [
                    {
                        id: 'analytics_navs',
                        title: '',
                        items: [
                            {
                                id: 'my_app_analytics_overview',
                                title: 'Overview',
                                href: '/comparator/',
                            },
                        ],
                    },
                ],
            },
        ];
        const sidebar = shallow(
            <Sidebar home="/home" platforms={platforms} onClickNav={spy} />
        );
        expect(sidebar).toMatchSnapshot();

        sidebar.find('Platform').at(0).prop('onClick')();
        expect(spy).toHaveBeenCalled();
    });

    it('render sidebar without analytics subNavs', () => {
        const platforms = [
            {
                id: 'my_app_analytics',
                title: 'My app analytics',
                href: '/comparator/landing/',
                items: null,
            },
        ];
        platforms[0].items = null;
        const sidebar = shallow(
            <Sidebar
                current=""
                home="/home"
                platforms={platforms}
                isPremium={Boolean(true)}
                isCollapsable
            />
        );
        expect(sidebar).toMatchSnapshot();
    });
});
