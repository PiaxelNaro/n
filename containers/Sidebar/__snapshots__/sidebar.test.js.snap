// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Container> render container 1`] = `
<container__Wrapper>
  <container__Main
    className="AASidebar "
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <container__LogoLink
      className=""
      href="/home"
    >
      <container__LogoMini
        className=""
      >
        <Logo__Logo
          mini={true}
        />
      </container__LogoMini>
      <container__LogoDefault
        className=""
      >
        <Logo__Logo />
      </container__LogoDefault>
    </container__LogoLink>
    <container__Content
      className=""
    >
      test
    </container__Content>
  </container__Main>
</container__Wrapper>
`;

exports[`<Container> render container 2`] = `
<container__Wrapper
  isCollapsable={true}
>
  <container__Main
    className="AASidebar is-collapsable"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <container__LogoLink
      className="is-collapsable"
      href="/home"
    >
      <container__LogoMini
        className="is-collapsable"
      >
        <Logo__Logo
          mini={true}
          premium={true}
        />
      </container__LogoMini>
      <container__LogoDefault
        className="is-collapsable"
      >
        <Logo__Logo />
        <container__PremiumText>
          PREMIUM
        </container__PremiumText>
      </container__LogoDefault>
    </container__LogoLink>
    <container__Content
      className="is-collapsable"
    >
      test
    </container__Content>
  </container__Main>
  <container__Mask
    className="is-collapsable"
  />
</container__Wrapper>
`;

exports[`<Container> render container wrapper 1`] = `
<div
  className="container__Wrapper-wqy56f-0 iYknXe"
>
  <container__Main
    className="AASidebar "
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <container__LogoLink
      className=""
      href="home"
    >
      <container__LogoMini
        className=""
      >
        <Logo__Logo
          mini={true}
        />
      </container__LogoMini>
      <container__LogoDefault
        className=""
      >
        <Logo__Logo />
      </container__LogoDefault>
    </container__LogoLink>
    <container__Content
      className=""
    >
      test
    </container__Content>
  </container__Main>
</div>
`;

exports[`<Container> render container wrapper 2`] = `
<div
  className="container__Wrapper-wqy56f-0 iYknXe"
>
  <container__Main
    className="AASidebar "
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <container__LogoLink
      className=""
      href="home"
    >
      <container__LogoMini
        className=""
      >
        <Logo__Logo
          mini={true}
        />
      </container__LogoMini>
      <container__LogoDefault
        className=""
      >
        <Logo__Logo />
      </container__LogoDefault>
    </container__LogoLink>
    <container__Content
      className=""
    >
      test
    </container__Content>
  </container__Main>
</div>
`;

exports[`<Link> render sidebar link 1`] = `
<link__Wrapper
  className=""
  onClick={[Function]}
>
  <FlexGrid
    gutter="2px"
  >
    <div />
    <FlexGrid.Item
      flex={1}
    >
      <FlexGrid>
        <link__Version />
      </FlexGrid>
    </FlexGrid.Item>
  </FlexGrid>
</link__Wrapper>
`;

exports[`<Link> render sidebar link 2`] = `
<link__Wrapper
  className="is-active"
  onClick={null}
>
  <FlexGrid
    gutter="2px"
  >
    <div />
    <FlexGrid.Item
      flex={1}
    >
      <FlexGrid>
        <link__Version>
          beta
          new
        </link__Version>
      </FlexGrid>
    </FlexGrid.Item>
    <MdLockOutline
      size="14"
    />
    <AAIcon
      name="aa-connections"
    />
  </FlexGrid>
</link__Wrapper>
`;

exports[`<NavGroup> render NavGroup with title 1`] = `
<platform__NavGroupWrapper
  className="AASidebar-group"
>
  <platform__NavGroupTitle
    className="AASidebar-group-title"
    ellipsis={true}
  >
    App Store Rankings
  </platform__NavGroupTitle>
  <Link
    isActive={false}
    onClick={[Function]}
  >
    <platform__GroupItem
      ellipsis={true}
    >
      Overview
    </platform__GroupItem>
  </Link>
</platform__NavGroupWrapper>
`;

exports[`<NavGroup> render NavGroup without header 1`] = `
<platform__NavGroupWrapper
  className="AASidebar-group"
>
  <Link
    isActive={false}
    onClick={[Function]}
  >
    <platform__GroupItem
      ellipsis={true}
    >
      Overview
    </platform__GroupItem>
  </Link>
</platform__NavGroupWrapper>
`;

exports[`<Platform> render 1`] = `
<platform__Wrapper
  className="AASidebar-platform "
>
  <Link
    className="AASidebar-platform-title"
    icon={
      <AAIcon
        name="aa-caret-down"
      />
    }
    onClick={[Function]}
  >
    <platform__PlatformTitle
      ellipsis={true}
    />
  </Link>
  <platform__Content
    className="AASidebar-platform-content "
  >
    <NavGroup
      current="my_app_analytics_overview"
      items={
        Array [
          Object {
            "href": "/comparator/",
            "id": "my_app_analytics_overview",
            "title": "Overview",
          },
        ]
      }
      onClick={[Function]}
      title=""
    />
  </platform__Content>
</platform__Wrapper>
`;

exports[`<Sidebar> render sidebar 1`] = `
<Container
  isCollapsable={false}
  isPremium={false}
>
  <Link
    icon={
      <AAIcon
        name="aa-home"
      />
    }
    isActive={true}
  >
    <Text__Text
      ellipsis={true}
    />
  </Link>
  <Platform
    navGroup={
      Array [
        Object {
          "id": "analytics_navs",
          "items": Array [
            Object {
              "href": "/comparator/",
              "id": "my_app_analytics_overview",
              "title": "Overview",
            },
          ],
          "title": "",
        },
      ]
    }
    onClick={[Function]}
    title="My app analytics"
  />
</Container>
`;

exports[`<Sidebar> render sidebar without analytics subNavs 1`] = `
<Container
  isCollapsable={true}
  isPremium={true}
>
  <Link
    icon={
      <AAIcon
        name="aa-home"
      />
    }
    isActive={false}
  >
    <Text__Text
      ellipsis={true}
    />
  </Link>
  <Link
    icon={<AAIcon />}
    isActive={false}
    link="/comparator/landing/"
  >
    <Text__Text
      ellipsis={true}
      textTransform="uppercase"
    >
      My app analytics
    </Text__Text>
  </Link>
</Container>
`;
